crates/rspack_binding_values/src/clean_options.rs

```diff
@@ -0,0 +1,67 @@
+use napi::bindgen_prelude::FromNapiValue;
+use napi::Either;
+use napi_derive::napi;
+use rspack_core::CleanOptions;
+use rspack_napi::napi;
+
+/// File clean options
+///
+/// A file filter is an option whether the file should be kept after clean up
+///
+/// TS Type:
+///
+/// ```typescript
+/// // in the future, we should support the following types, just like webpack
+/// // type CleanOptions = boolean | { dry?: boolean, keep?: RegExp | string | ((filename: string) => boolean) }
+///
+/// type CleanOptions = boolean | { keep?: string }
+/// ```
+#[derive(Debug)]
+pub struct RawCleanOptions(Either<bool, JsCleanOptions>);
+
+impl FromNapiValue for RawCleanOptions {
+  unsafe fn from_napi_value(
```

Comment:
This `unsafe` is just using to convert `either`. You can try to handle this in `packages/rspack/src/config/normalization.ts`, just convert `clean: true/false` to a `RawCleanOptions` struct which can be used on the Rust side directly. Simple Rust API is easy for native accessors to understand.

Reply:
It will make the problem worse...

I think we'd better get rid of this struct and just use `Either` directly. I agree with the previous review now.